### PR TITLE
Dob dedup issue 360

### PR DESCRIPTION
**Pull request by td928**
Refer to https://github.com/NYCPlanning/db-knownprojects/issues/360 for the full extent of issue this branch addresses.

### CODE REVIEW

File: sql/project_record_ids.sql [hunk 1]
```diff
@@ -22,11 +22,35 @@ OUTPUTS:
 
 -- Copy pre-DOB match _project_record_ids into project_record_ids;
 DROP TABLE IF EXISTS project_record_ids;
+DROP TABLE IF EXISTS verified_matches;
```
Comment (AmandaDoyle) [inline sql/project_record_ids.sql:25]: This is fine to keep in, but `verified_matches` is only a temporary table created with a `WITH` statement so there should never be a table that needs dropping.

**Review by AmandaDoyle — COMMENTED** [review]
Need to discuss.

**Comment by td928** [comment]
@mbh329 `2021K0221` is record that I would check for the dob clustering is working properly in `project_record_ids`

**Comment by mbh329** [comment]
Will need to merge dev into this branch as well once the other PR #372  is approved @td928

**Comment by td928** [comment]
> Will need to merge dev into this branch as well once the other PR #372 is approved @td928

I think I already did. This branch should be up to date with dev.

File: python/dedup_units.py [hunk 2]
```diff
@@ -42,6 +42,10 @@ def resolve_all_projects(df):
     resolved = df.groupby(["project_id"], as_index=False).apply(resolve_project)
     resolved = resolved[["project_id", "source", "units_gross", "units_net", "record_id"]]
     print(resolved.head())
+
+    # removing duplicate record after
```
Comment (mbh329) [inline python/dedup_units.py:46]: I think the comment here should be focused on why we want to remove duplicate records - anytime we can write out the logic in layman's terms it will help the engineer figure out what's going on

File: sql/project_record_ids.sql [hunk 3]
```diff
@@ -25,8 +25,31 @@ DROP TABLE IF EXISTS project_record_ids;
...
+
+
+/* Add stand-alone projects. This includes unmatched residential DOB projects, 
+as well as projects from sources that were excluded 
+from the non-DOB match process. */
+INSERT INTO project_record_ids
+SELECT array[]::text[]||record_id as project_record_ids
+FROM (
+	SELECT record_id::text from combined
+	WHERE no_classa = '0' OR no_classa IS NULL
```
Comment (mbh329) [inline sql/project_record_ids.sql:37]: `no_classa` units is the number of Class A units?

File: python/dedup_units.py (hunk 2 above)
Comment (td928) [reply python/dedup_units.py:46]: excellent question. Almost forgot all about this after a year. Trying to remember and will write better descriptions

File: sql/project_record_ids.sql (hunk 3 above)
Comment (td928) [reply sql/project_record_ids.sql:37]: it's binary for '0' as having no class a units and '1' for having any units

Comment (mbh329) [reply sql/project_record_ids.sql:37]: Got it, thanks so its actually "no class A units"

File: python/dedup_units.py (hunk 2 above)
Comment (td928) [reply python/dedup_units.py:46]: this turns out to be a more significant issue. I think this thing I did might not be legal and I need to look more upstream for a better solution. Stay tuned for this @mbh329

File: sql/project_record_ids.sql [hunk 4]
```diff
@@ -25,8 +25,32 @@ DROP TABLE IF EXISTS project_record_ids;
...
+
+/* Add stand-alone projects. This includes unmatched residential DOB projects, 
+as well as projects from sources that were excluded 
+from the non-DOB match process. */
+INSERT INTO project_record_ids
+SELECT array[]::text[]||record_id as project_record_ids
+FROM (
+	SELECT record_id::text from combined
+	WHERE (no_classa = '0' OR no_classa IS NULL) 
+	AND source <> 'DOB'
```
Comment (td928) [inline sql/project_record_ids.sql:38]: Hey @mbh329  see the adjustments here I made for addressing your comments about that drop duplicates question you had. It turns out I added bunch of irrelevant dob records here earlier by not excluding the non-cluster DOB records. That's why I had to do the drop duplicates the end but that is not a real solution for the problem. With this new logic it should be good to go. Let me know if you still have any question about this

**Review by mbh329 — APPROVED** [review]
LGTM
